File: Simulating and Visualizing Character Movement on a Cartesian Plane.py
import matplotlib.pyplot as plt

# Define the initial position and the path of the character
initial_position = (0, 0)
path = [
    (2, -1),
    (4, -3),
    (3, -4),
    (0, 0)
]

# Extract x and y coordinates from the path
x_coords = [initial_position[0]] + [pos[0] for pos in path]
y_coords = [initial_position[1]] + [pos[1] for pos in path]

# Create a plot
plt.figure(figsize=(8, 8))

# Plot the path
plt.plot(x_coords, y_coords, 'ro-')  # 'ro-' means red color, circle markers, solid line

# Annotate each point
for i, (x, y) in enumerate(zip(x_coords, y_coords)):
    plt.text(x, y, f'({x}, {y})', fontsize=12, ha='right')

# Set the x and y axis labels
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

# Set the x and y axis limits
plt.xlim(-1, 5)
plt.ylim(-5, 1)

# Draw grid
plt.grid(True)

# Title
plt.title('Character Movement in "The Sims" on a 2D Cartesian Plane')

# Draw axes
plt.axhline(0, color='black',linewidth=0.5)
plt.axvline(0, color='black',linewidth=0.5)

# Show the plot
plt.show()
t

File: Simulating and Visualizing Character Movement on a Cartesian Plane.cs
using System;

class Program
{
    static void Main()
    {
        // Coordinates of the point
        int xPoint = 3;
        int yPoint = 4;

        // Simulate the character's movement in "The Sims"
        // Starting at (0, 0), moving to (2, -1), (4, -3), (3, -4), and back to (0, 0)
        int[,] path = new int[,]
        {
            {0, 0},
            {2, -1},
            {4, -3},
            {3, -4},
            {0, 0}
        };

        // Plot the point (3, 4)
        Console.WriteLine("Plotting the point (3, 4) on a 2D Cartesian plane:");
        Console.WriteLine($"Point: ({xPoint}, {yPoint})");

        // Simulate the movement
        Console.WriteLine("\nSimulating character movement in 'The Sims':");
        for (int i = 0; i < path.GetLength(0); i++)
        {
            Console.WriteLine($"Step {i + 1}: ({path[i, 0]}, {path[i, 1]})");
        }

        // Visual representation
        Console.WriteLine("\nVisual representation (X, Y):");

        // Creating a grid for visualization
        int gridSize = 10; // Define grid size
        char[,] grid = new char[gridSize, gridSize];

        // Initialize the grid with dots
        for (int i = 0; i < gridSize; i++)
        {
            for (int j = 0; j < gridSize; j++)
            {
                grid[i, j] = '.';
            }
        }

        // Mark the point (3, 4) on the grid
        grid[gridSize / 2 - yPoint, gridSize / 2 + xPoint] = 'P';

        // Mark the path on the grid
        for (int i = 0; i < path.GetLength(0); i++)
        {
            int x = path[i, 0];
            int y = path[i, 1];
            grid[gridSize / 2 - y, gridSize / 2 + x] = 'C';
        }

        // Print the grid
        for (int i = 0; i < gridSize; i++)
        {
            for (int j = 0; j < gridSize; j++)
            {
                Console.Write(grid[i, j] + " ");
            }
            Console.WriteLine();
        }
    }
}
